lib/Sema/TypeCheckGeneric.cpp

```diff
@@ -77,6 +80,29 @@ TypeChecker::gatherGenericParamBindingsText(
   return result.str().str();
 }
 
+// An alias to avoid repeating the `SmallVector`'s size parameter.
+using CollectedOpaqueReprs = SmallVector<OpaqueReturnTypeRepr *, 2>;
+
+/// Walk `repr` recursively, collecting any `OpaqueReturnTypeRepr`s.
+static CollectedOpaqueReprs collectOpaqueReturnTypeReprs(TypeRepr *repr) {
+  class Walker : public ASTWalker {
+    CollectedOpaqueReprs &Reprs;
+
+  public:
+    explicit Walker(CollectedOpaqueReprs &reprs) : Reprs(reprs) {}
+
+    bool walkToTypeReprPre(TypeRepr *repr) override {
+      if (auto opaqueRepr = dyn_cast<OpaqueReturnTypeRepr>(repr))
+        Reprs.push_back(opaqueRepr);
+      return true;
+    }
+  };
+
+  CollectedOpaqueReprs reprs;
```

Comment:
Can this use your new findIf() method?

Reply:
I don't think so. Unless I'm missing something, I need to examine all of the opaque type reps. findIf() functions like || and bails after it hits the first type repr satisfying the given predicate.